x-pack/plugins/actions/server/builtin_action_types/swimlane/service.ts

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Logger } from '@kbn/logging';
+import axios from 'axios';
+
+import { ActionsConfigurationUtilities } from '../../actions_config';
+import { getErrorMessage, request } from '../lib/axios_utils';
+import { getBodyForEventAction } from './helpers';
+import {
+  CreateCommentParams,
+  CreateRecordParams,
+  ExternalService,
+  ExternalServiceCredentials,
+  ExternalServiceIncidentResponse,
+  MappingConfigType,
+  ResponseError,
+  SwimlanePublicConfigurationType,
+  SwimlaneRecordPayload,
+  SwimlaneSecretConfigurationType,
+  UpdateRecordParams,
+} from './types';
+import * as i18n from './translations';
+
+const createErrorMessage = (errorResponse: ResponseError | null | undefined): string => {
+  if (errorResponse == null) {
+    return '';
```

Comment:
It would probably be slightly better to return "unknown" or similar here, instead of an empty string.  Or instead of passing the errorResponse, pass the error, and if there's no response data, use the `error.message` as a fallback.